s3blobs/src/play/modules/s3blobs/S3Blob.java

```diff
@@ -116,16 +118,16 @@ public int hashCode(Object o) throws HibernateException {
 	}
 
 	@Override
-	public Object nullSafeGet(ResultSet rs, String[] names, Object o) throws HibernateException, SQLException {
-		String val = StringType.INSTANCE.nullSafeGet(rs, names[0]);
-		if (val == null || val.length() == 0 || !val.contains("|")) {
-			return new S3Blob();
+	public Object nullSafeGet(ResultSet rs, String[] names, SessionImplementor owner, Object o) throws HibernateException, SQLException {
+		String val = StringType.INSTANCE.nullSafeGet(rs, names[0], owner);
+		if (val == null || val.length() == 0 || !val.contains("|") || val.equals("null|null")) {
+			return null;
```

Comment:
If no value has been set for this object and it's null in the database then null should be returned. Returning a new S3Blob results in an error occurring while calling .get on the object at a later stage. For example if User.photo is an S3Blob , but photo is optional, in the application you'd need to check if it's null to know if a photo has actually been set for that user.